sebex/cmd/release.py

```diff
@@ -30,25 +31,70 @@ def status():
     else:
         success('There is no release pending at this moment, feel free to start one.')
 
+def gather_input() -> dict[Version, ProjectHandle]:
+    bumps = {}
+    log("hit enter when done", color='yellow')
+    while True:
+        value = click.prompt("Project", default="", show_default=False)
+        if value == "":
+            break
+        project = valid_project(value)
+        if project is None:
+            continue
+        # TODO at this moment all packages are bumped by one minor version
+        # TODO it would be better to specify different release versions for different packages
+        # while True:
+        #     value = click.prompt("Version")
+        #     version = valid_version(value)
+        #     if version is not None:
+        #         bumps[project] = version
+        #         break
+        bumps[project] = None # replace this line with commented code above to use version
+    return bumps
+
+# def valid_version(value: str) -> Optional[Version]:
+#     try:
+#         return Version.parse(value)
+#     except ValueError:
+#         print(f'{value!r} is not a valid version')
+#         return None
+
+def valid_project(value: str) -> Optional[ProjectHandle]:
+    try:
+        handle = ProjectHandle.parse(value)
+    except ValueError:
+        print(f'{value!r} is not a valid project name')
+    manifest = Manifest.open()
+    for repo_manifest in manifest.iter_repositories():
+        for project in repo_manifest.project_handles():
+            if project == handle:
+                return project
+    print(f'Unknown project {handle}')
+    return None
 
 @release.command()
-@click.option('--project', type=PROJECT, required=True, prompt=True)
-@click.option('--version', type=VERSION, required=True, prompt=True)
 @click.option('--dry', is_flag=True,
               help='Print what would be done, but do not persist the generated plan.')
-def plan(project: ProjectHandle, version: Version, dry: bool):
+def plan(dry: bool):
     """
-    Prepare release plan for managed package.
+    Prepare release plan for managed packages.
     """
 
+    # gather project names of packages to be released
+    bumps = gather_input()
+    assert len(bumps) > 0
+
     if not dry:
         if ReleaseState.exists():
             rel = ReleaseState.open()
             fatal(f'Release "{rel.codename()}" is already running.',
                   'Please finish it before creating new one.')
 
-    database, graph = analyze()
-    rel = ReleaseState.plan(project, version, database, graph)
+    database, graph = analyze(bumps)
+    # creating the v_project happens under `analyze` in `AnalysisDatabase._add_virtual_root`
+    v_project = database.get_project_by_package(VIRTUAL_NODE)
+    # bump v_project from 0.1.0 -> 0.2.0 thereby bumping all dependent packages
+    rel = ReleaseState.plan(v_project, Version(0, 2, 0), database, graph)
```

Comment:
This plan version 0.2.0 thingy smell a bit like logic smearing. Maybe you could move most of this logic closer to VIRTUAL_NODE definition? At least consider moving `Version(0, 2, 0)` magic there. Or move everything related to separate file.